src/core/utils/AlbedoNormalPass.js

```diff
@@ -0,0 +1,224 @@
+		this.albedoMaterials = new Map();
+		this.useWorldSpaceNormals = useWorldSpaceNormals;
+		//debugging
+		//this.debugNormalMaps = true;
+
+		// RT
+		this.renderTarget = new THREE.WebGLRenderTarget( 1, 1, {
+			samples: 16,
+			count: 2
+		} );
+		this.renderTarget.texture.colorSpace = THREE.SRGBColorSpace;
+
+		this.albedoNormalMaterial = new THREE.ShaderMaterial( {
+			uniforms: {
+				diffuseMap: { value: null },
+				normalMap: { value: null },
+				color: { value: new THREE.Color( 1, 1, 1 ) },
+				useWorldSpaceNormals: { value: this.useWorldSpaceNormals },
+				useDiffuseMap: { value: false },
+				useNormalMap: { value: false },
+				debugNormalMaps: { value: this.debugNormalMaps },
+				normalScale: { value: 1 }
+			},
+			vertexShader: /* glsl */`
+                varying vec2 vUv;
+                varying vec3 vViewPosition;
+                varying vec3 vNormal;
+                varying vec3 vWorldNormal;
+                varying vec3 vTangent;
+                varying vec3 vBitangent;
+
+                void main() {
+                    vUv = uv;
+                    vec4 mvPosition = modelViewMatrix * vec4(position, 1.0);
+                    gl_Position = projectionMatrix * mvPosition;
+                    vViewPosition = -mvPosition.xyz;
+                    vNormal = normalize(normalMatrix * normal);
+                    vWorldNormal = normalize((modelMatrix * vec4(normal, 0.0)).xyz);
+                    
+                    #ifdef USE_TANGENT
+                        vTangent = normalize(normalMatrix * tangent.xyz);
+                        vBitangent = normalize(cross(vNormal, vTangent) * tangent.w);
+                    #endif
+                }
+            `,
+			fragmentShader: /* glsl */`
+                uniform vec3 color;
+                uniform sampler2D diffuseMap;
+                uniform sampler2D normalMap;
+                uniform bool useWorldSpaceNormals;
+                uniform bool useDiffuseMap;
+                uniform bool useNormalMap;
+                uniform bool debugNormalMaps;
+				uniform float normalScale;
+
+                varying vec2 vUv;
+                varying vec3 vViewPosition;
+                varying vec3 vNormal;
+                varying vec3 vWorldNormal;
+                varying vec3 vTangent;
+                varying vec3 vBitangent;
+
+                layout(location = 0) out vec4 gAlbedo;
+                layout(location = 1) out vec4 gNormal;
+
+                void main() {
+                    // Albedo
+                    vec3 albedo;
+                    if (useDiffuseMap) {
+                        vec4 diffuseColor = texture(diffuseMap, vUv);
+                        albedo = diffuseColor.rgb * color;
+                    } else {
+                        albedo = color;
+                    }
+                    gAlbedo = vec4(albedo, 1.0);
+
+                    // Normal
+                    vec3 normal;
+                    if (useNormalMap) {
```

Comment:
We should be able to reuse a lot of the three.js build in shader #includes to generate all these values without so much code.